djstripe/stripe_objects.py

```diff
@@ -262,6 +303,54 @@ def _stripe_object_to_invoice(cls, target_cls, data):
         return target_cls._get_or_create_from_stripe_object(data, "invoice")[0]
 
     @classmethod
+    def _stripe_object_to_invoice_items(cls, target_cls, data, invoice):
+        """
+        Retrieves InvoiceItems for an invoice.
+
+        If the invoice item doesn't exist already then it is created.
+
+        If the invoice is an upcoming invoice that doesn't persist to the
+        database (i.e. ephermeral) then the invoice items are also not saved.
+
+        :param target_cls: The target class to instantiate per invoice item.
+        :type target_cls: ``StripeInvoiceItem``
+        :param data: The data dictionary received from the Stripe API.
+        :type data: dict
+        """
+        lines = data.get("lines")
+        if not lines:
+            return []
+
+        items = []
+        for line in lines.get("data", []):
+            if invoice.stripe_id:
+                save = True
+                line.setdefault("invoice", invoice.stripe_id)
+
+                if line.get("type") == "subscription":
+                    # Lines for subscriptions need to be keyed based on invoice and
+                    # subscription, because their id is *just* the subscription
+                    # when received from Stripe. This means that future updates to
+                    # a subscription will change previously saved invoices - Doing
+                    # the composite key avoids this.
+                    if not line["id"].startswith(invoice.stripe_id):
+                        line["id"] = "{invoice_id}-{sub_id}".format(
+                            invoice_id=invoice.stripe_id,
+                            sub_id=line["id"])
+            else:
+                # Don't save invoice items for ephemeral invoices
+                save = False
+
+            line.setdefault("customer", invoice.customer.stripe_id)
+            line.setdefault("date", int(invoice.date.strftime("%s")))
```

Comment:
Yep, that's right - Looks like we can borrow from Django instead: 

```
from django.utils import dateformat
line.setdefault("date", dateformat.format(invoice.date, 'U'))
```

I looked at the implementation and it's pretty similar to your suggestion with the main difference that it uses `utctimetuple` if the datetime if timezone aware:

```
    def U(self):
        "Seconds since the Unix epoch (January 1 1970 00:00:00 GMT)"
        if isinstance(self.data, datetime.datetime) and is_aware(self.data):
            return int(calendar.timegm(self.data.utctimetuple()))
        else:
            return int(time.mktime(self.data.timetuple()))
```

Thoughts?